### ISSUE TITLE
Code solution for Sql-2-R is wrong

### ISSUE THREAD

**Issue by ethanwhite** [datacarpentry/semester-biology#283]
```
SELECT species_id, sex, AVG(hindfoot_length), AVG(weight)
FROM surveys
WHERE species_id IS NOT NULL AND sex IS NOT NULL
GROUP BY species_id, sex;
```
Yields 49 rows.

**Comment by ethanwhite** [comment]
I think the issue here results from the fact that the code solution loops over all possible species sex combinations instead of just using a GROUP BY. The GROUP BY will be the more common approach.

**Comment by brymz** [comment]
Wow. Of course, `GROUP BY` is the better solution for this question. I am revising the exercise and solution to make the direct connection to `GROUP BY`.

The bigger problem... I constructed this exercise with the goal of integrating a sequence of queries with 'R Fundamentals' [Step 1. make a query; Step 2.: generalize the query in a custom function; Step 3. automate process with a `for` loop]. That no longer happens with the `GROUP BY` solution (*much more elegant*). Do you think that R-SQL integration happens sufficiently throughout the remainder of the course? or do we want to come up with another R-SQL problem / revise 'R-SQL 3' to meet these needs?

**Comment by ethanwhite** [comment]
> Do you think that R-SQL integration happens sufficiently throughout the remainder of the course? or do we want to come up with another R-SQL problem / revise 'R-SQL 3' to meet these needs?

I don't have any strong opinion on this at the moment.

This was closed by #284.
